taglib/toolkit/tbytevector.cpp

```diff
@@ -49,7 +49,7 @@
 //
 // http://www.informit.com/isapi/product_id~{9C84DAB4-FE6E-49C5-BB0A-FB50331233EA}/content/index.asp
 
-#define DATA(x) (&(x->data->data[0]))
+#define DATA(x) (&(*(x)->data)[0])
```

Comment:
I'd say at least remove them where you simply do:

DATA(d)[someindex]

to

d->data[someindex]